File: game/src/main/java/com/scottlogic/hackathon/game/GameResult.java
package com.scottlogic.hackathon.game;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface GameResult {
    UUID getId();
    List<PhaseResult> getPhaseResults();
    GameMap getMap();
    Set<Position> getOutOfBoundPositions();
    CutoffCondition getCutoffCondition();
}

File: ui/src/interfaces/GameResult.ts
import { CutoffCondition } from '~/enums/CutoffCondition';

import { Game } from './Game';
import { PhaseResult } from './PhaseResult';
import { SpawnPointData } from './SpawnPointData';

export interface GameResult {
  cutoffCondition: CutoffCondition;
  game: Game;
  id: string;
  phaseResults: PhaseResult[];
  spawnPoints: SpawnPointData[];
}
